onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder.cc

```diff
@@ -1195,14 +1221,17 @@ Status BatchNormalizationOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_bu
   const auto tensor_a_name = model_builder.GetUniqueName(node.Name() + input + "_imm_a");
   const auto tensor_b_name = model_builder.GetUniqueName(node.Name() + input + "_imm_b");
   const auto tensor_imm_product_name = model_builder.GetUniqueName(node.Name() + input + "_imm_mul");
-  Shape tensor_a_dimen;
+  Shape tensor_a_dimen = {size};
 
   bool input_is_nhwc = model_builder.IsOperandNHWC(input);
   bool output_is_nhwc = input_is_nhwc;
-  if (input_is_nhwc)
-    tensor_a_dimen = {size};
-  else                              // input is nchw
-    tensor_a_dimen = {size, 1, 1};  // {C, H, W}
+
+  if (!input_is_nhwc) {
+    // input shape is {N, C, [H, W]}, so tensor_a/b's shape is {C, [H, W]}
+    const auto input_rank = shaper[input].size();
```

Comment:
Not quite clear 1 is pushed for the H and W dims if input is nchw so a comment explaining why would be helpful.